ext2/pc-ata.c

```diff
@@ -440,7 +440,7 @@ static int ata_init_bus(struct ata_bus *ab)
 			ab->ac[i].devices[j].drive = j;
 			ab->ac[i].devices[j].signature = ab->ac[i].devices[j].info.general_config;
 			ab->ac[i].devices[j].capabilities = ab->ac[i].devices[j].info.capabilities_1;
-			ab->ac[i].devices[j].command_sets = *((uint32_t*)(&(ab->ac[i].devices[j].info.commands1_sup)));
+			ab->ac[i].devices[j].command_sets = ((uint32_t)(ab->ac[i].devices[j].info.commands1_sup) << 16) + ab->ac[i].devices[j].info.commands2_sup;
```

Comment:
Should be commands2_sup << 16 | commands1_sup or result will have flipped byte order.
Also using bitwise OR rather than addition would be a little bit cleaner imho

Reply:
Fixed as suggested.